client/state/products-list/selectors/plan-slug-to-plan-product.js

```diff
@@ -1,26 +0,0 @@
-import {
-	applyTestFiltersToPlansList,
-	applyTestFiltersToProductsList,
-	getPlan,
-	getProductFromSlug,
-	objectIsProduct,
-} from '@automattic/calypso-products';
-
-/**
- * Computes a plan object and a related product object based on plan slug/constant
- *
- * @param {Array[]} products A list of products
- * @param {string} planOrProductSlug Plan constant/slug
- * @returns {object} Object with a related plan and product objects
- */
-export const planSlugToPlanProduct = ( products, planOrProductSlug ) => {
-	const plan = getPlan( planOrProductSlug ) ?? getProductFromSlug( planOrProductSlug );
-	const constantObj = objectIsProduct( plan )
-		? applyTestFiltersToProductsList( plan )
```

Comment:
The line above this explicitly says that `plan` is an object but `applyTestFiltersToProductsList()` only operates on strings. My best guess is this did not explode because of some happy coincidence.